File: C#/Domain/RollDiceGame.cs
namespace Domain
{
    public class RollDiceGame
    {
        private int playersCount;

        public void AddPlayer(Player player)
        {
            if (playersCount == 6)
            {
                throw new TooManyPlayersException();
            }
            playersCount++;
        }

        public void RemovePlayer(Player player)
        {
            playersCount--;
        }
    }
}

File: Flutter/Unit-01/lib/roll_dice_game.dart
library casino;

import 'dice.dart';
import 'player.dart';
import 'exceptions.dart';

class RollDiceGame {
  final List<Player> _players = <Player>[];
  late Dice _dice;

  RollDiceGame(Dice dice) {
    _dice = dice;
  }

  addPlayer(Player player) {
    if (_players.length == 6) throw TooManyPlayersException();
    _players.add(player);
  }

  removePlayer(Player player) {
    _players.remove(player);
  }

  play() {
    var winningScore = _dice.roll();
    for (var player in _players) {
      var bet = player.getBet();
      if (bet == null) continue;

      if (bet.score == winningScore) {
        player.win(bet.chips * 6);
      }
      else {
        player.lose(bet.chips);
      }
    }
  }
}